### PR TITLE
Restore build and and basic features of SiteSettings - in progress.

### PR DESCRIPTION
**Pull request by mean2me**
This PR is a not definitive to restore all functionalities of SiteSettings.
There are UI issues related to Collapsible component and some hidden component in Languages and Privacy settings.

### CODE REVIEW

**Comment by mitchelsellers** [comment]
@valadas @mtrutledge Can we get a review from you guys?

**Review by mtrutledge — CHANGES_REQUESTED** [review]
Let's make sure we remove the old dnn components from the package.json and regenerate the yarn.lock file.

Also, I noticed going through the components that you removed `componentDidUpdate` in the files, but dont see that logic moved any where else. Is there a reason for this? This particular lifecycle method is still valid. The ones being deprecated are `componentWillMount`, `componentWillUpdate`, and `componentWillReceiveProps`.

**Comment by mean2me** [comment]
Hello @mtrutledge .
The reason why I had to remove all those **componentDidUpdate** is because they have a **setState** call inside and it created a recursive issue. Basically it's the main reason why we couldn't see SiteSettings on the screen but there is a bit more work to do. 
I sent this PR to avoid a too long one and to separate build problems and functional problems.
Also, I don't like the way the UI components are initialized. They are mounted with empty data, then they call data in **componentDidUpdate**. It means we have many redundant rendering of the UI and potential null pointer errors. We have Redux and components should be very lightweight and just wait for Props update from the top. I think SiteSettings (and probably other components) will need a deep refactoring if we want to be 100% sure that it works fine. IMHO.

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/package.json
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/package.json]: @dnnsoftware/dnn-react-common has everything, so dnn-global-styles and dnn-webpack-externals should be removed

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/basicSearchSettings/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/basicSearchSettings/index.jsx:57]: I am also concerned about this removal without it being implemented another way. Although I also don't like the original implementation, I am scared to just remove it without seing it reimplemented another way. Also since the tab that has this does not load currently, I have no way to property test the effects of this change.

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/defaultPagesSettings/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/defaultPagesSettings/index.jsx:74]: Same here, I do not see this reimplemented another way

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/ignoreWords/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/ignoreWords/index.jsx:92]: Same here, I do not undertand just removing it without reimplemented

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/index.jsx:79]: Same here

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/languages/languageEditor/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/languages/languageEditor/index.jsx:48]: same here, removed without a replacement solution

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/languages/languageEditor/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/languages/languageEditor/index.jsx]: This method is removed but it is used further down in the file, this will fail.

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/languages/languageEditor/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/languages/languageEditor/index.jsx:196]: Why this removal?

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/messagingSettings/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/messagingSettings/index.jsx:99]: Same here, removed without alternate solution

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/moreSettings/htmlEditorManager/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/moreSettings/htmlEditorManager/index.jsx:24]: Same, removed without alternate solution

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/moreSettings/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/moreSettings/index.jsx:51]: Same, removed without alternate solution

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/privacySettings/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/privacySettings/index.jsx:74]: Same here, removed without alternate solution

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/privacySettings/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/privacySettings/index.jsx]: I cannot properly test this now since this tab does not load, can you explain the motivation behind this change?

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/profileSettings/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/profileSettings/index.jsx:58]: Same, removed without alternate solution

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/profileSettings/profileProperties/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/profileSettings/profileProperties/index.jsx:64]: Same, removed without alternate solution

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/profileSettings/profileProperties/profilePropertyEditor/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/profileSettings/profileProperties/profilePropertyEditor/index.jsx:61]: Same here, removed without alternate solution

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/profileSettings/profileProperties/profilePropertyRow/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/profileSettings/profileProperties/profilePropertyRow/index.jsx:17]: This Collapse is not the component but a function passed down as props, this one should not get renamed to Collapsible

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/siteAliasSettings/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/siteAliasSettings/index.jsx:56]: Removed without alternate solution

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/siteAliasSettings/siteAliases/siteAliasEditor/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/siteAliasSettings/siteAliases/siteAliasEditor/index.jsx:43]: Removed without alternate solution

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/synonymsGroups/index.jsx
Comment (valadas) [inline Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/synonymsGroups/index.jsx:92]: Removed without alternate replacement

**Review by valadas — CHANGES_REQUESTED** [review]
The replacement of components looks pretty fine to me but I also have concerns with methods removed without alternate solutions. I cannot test this fully now because some pages do not load, so I see 2 solutions: 
A: Either we undo the concerns and merge was is good then do a second PR to complete this.
B: We need to continue working on this PR until everything works as before and property test.

But right now I would be concerned about loosing functionality with those methods removed without a testable alternative that does the same job. FWIW I have no issue just using the unsafe_ methods for this initial round of cleanups just to bring things to a working state.

**Comment by mean2me** [comment]
@valadas @mitchelsellers I agree about your concern about methods removal. Actually I'm working on alternate replacement case by case. Let's consider this PR as a good discussion thread to understand more about what to do. I think in next two days I'll commit missing methods. It takes time because I'm restoring each feature in the best way as possible and sometimes it requires some more refactoring.

**Comment by mtrutledge** [comment]
This fix for basicSettings looks good. How many more of the componentDidUpdate functions are left?

**Comment by mean2me** [comment]
Hello @mtrutledge , I've just committed similar fixed for Privacy Settings and Site Alias Settings panels.
Now I'm gonna focus on Languages and I'm also waiting for latest changes in React.Common to be available.

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/basicSearchSettings/index.jsx
Comment (mean2me) [reply Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/basicSearchSettings/index.jsx:57]: Here it can be removed since it's redundant and misused. All we need already happens in **onSettingChange()** method.

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/defaultPagesSettings/index.jsx
Comment (mean2me) [reply Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/defaultPagesSettings/index.jsx:74]: It has been reintroduced and improved in following commits.

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/index.jsx
Comment (mean2me) [reply Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/index.jsx:79]: this method is actually useless. It has been removed and component has been refactored accordingly.

**Comment by mtrutledge** [comment]
Version 2.0.7 has been pushed to npm. I know there are a couple more pull requests coming for common as well. Once those are approved I can publish a new version.

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/siteAliasSettings/siteAliases/siteAliasEditor/index.jsx
Comment (mean2me) [reply Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/siteAliasSettings/siteAliases/siteAliasEditor/index.jsx:43]: Removed because redundant. Just a duplicate of **onSettingChange()** with redundant setState call.

File: Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/siteAliasSettings/index.jsx
Comment (mean2me) [reply Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/siteAliasSettings/index.jsx:56]: restored and improved.

**Comment by mean2me** [comment]
Hello @mtrutledge and @valadas ,
I checked all your comments and fixed everything accordingly. This PR is huge, it contains many changes split in several commits. Testing locally, all seems to be back to a stable situation now.
cc: @daguiler

**Comment by daguiler** [comment]
Let's use squash & merge

**Review by mtrutledge — APPROVED** [review]
Great work @mean2me !

**Review by daguiler — APPROVED** [review]

**Review by ohine — APPROVED** [review]

**Pull request merged**
